include/oneapi/dnnl/dnnl.hpp

```diff
@@ -2967,6 +2969,23 @@ struct memory : public handle<dnnl_memory_t> {
             reset(md);
         }
 
+        /// Constructs a memory descriptor describing a host-side
+        /// scalar to be directly passed as an argument to device-side
+        /// kernels. Intended to avoid memory transfer overhead on
+        /// primitive execution. The resulting descriptor cannot be
+        /// utilized to generate a memory object.
+        ///
+        /// @param adata_type Data type of the scalar.
+        /// @returns A memory descriptor for host-side scalar input.
+        desc(data_type adata_type) {
```

Comment:
I feel it's better to follow the sparse signature `static desc host_scalar(dt)` instead to have:
* A clear name;
* Untied hands if any extension will be needed;
* Confusion why just a data type ctor leads to a host-side memory and not anything else;
 
@mgouicem, any thoughts on this one?

Reply:
Done.